packages/react/src/_VisuallyHidden.tsx

```diff
@@ -1,29 +1,25 @@
-import styled from 'styled-components'
+import classes from './_VisuallyHidden.module.css'
+import {clsx} from 'clsx'
+import {BoxWithFallback} from './internal/components/BoxWithFallback'
 import type {SxProp} from './sx'
-import sx from './sx'
 
 interface Props {
   isVisible?: boolean
+  as?: React.ElementType
 }
 
-const VisuallyHidden = styled.span<Props & SxProp>`
-  ${({isVisible = false}) => {
-    if (isVisible) {
-      return sx
-    }
-
-    return `
-        position: absolute;
-        width: 1px;
-        height: 1px;
-        padding: 0;
-        margin: -1px;
-        overflow: hidden;
-        clip: rect(0, 0, 0, 0);
-        white-space: nowrap;
-        border-width: 0;
-    `
-  }}
-`
+function VisuallyHidden({
+  isVisible,
+  children,
+  as = 'span',
+  className,
+  ...rest
+}: Props & React.HTMLAttributes<HTMLSpanElement> & SxProp) {
```

Comment:
If `as` is offered then the given `React.HTMLAttributes` type here could be incorrect if they pass in something other than a `span`. I think this would need to be something like:

```tsx
```suggestion
function VisuallyHidden<As extends React.ElementType>({
  isVisible,
  children,
  as = 'span',
  className,
  ...rest
}: Props & { as?: As } & React.ComponentPropsWithoutRef<ElementType extends As ? As : 'span'> & SxProp) {
```
```